diff -r bc0814972646 -r cfda5e8995f2 tmpwatch.c
--- a/tmpwatch.c	Thu Oct 15 16:22:45 2009 +0200
+++ b/tmpwatch.c	Tue Dec 22 16:03:03 2009 +0100
@@ -227,12 +227,13 @@
     static char *const empty_environ[] = { NULL };
 
     int ret;
-    char dir[FILENAME_MAX];
+    char dir[2 + PATH_MAX];
     int pid;
 
     /* should we close all unnecessary file descriptors here? */
-    
-    snprintf(dir, sizeof(dir), "%s", filename);
+
+    /* Use "./" to protect against filenames starting with '-' */
+    snprintf(dir, sizeof(dir), "./%s", filename);
     pid = fork();
     if (pid == 0) {
 	execle(FUSER_PATH, FUSER_PATH, FUSER_ARGS, dir, NULL, empty_environ);
